web/js/modules/image-download/util.js

```diff
@@ -287,3 +289,53 @@ export function getPercentageFromPixel(maxDimension, dimension) {
 export function getPixelFromPercentage(maxDimension, percent) {
   return Math.round((percent / 100) * maxDimension);
 }
+
+/**
+ * Find if there are layers that cannot be downloaded
+ * @param {Array} visibleLayers
+ *
+ * @return {Bool}
+ */
+export function hasNonDownloadableVisibleLayer(visibleLayers) {
+  const notDownloadable = !!lodashFind(visibleLayers, { snapshot: false });
+  return notDownloadable;
+}
+/**
+ * Get string of layers to be removed if alert is accepted
+ * @param {Array} nonDownloadableLayers
+ *
+ * @return {String}
+ */
+export function getNamesOfNondownloadableLayers(nonDownloadableLayers) {
+  let names = '';
+  if (nonDownloadableLayers.length) {
+    nonDownloadableLayers.forEach((obj) => {
+      const str = names ? `, ${obj.title || obj.id}` : obj.title || obj.id;
+      names += str;
+    });
+  }
+  return names;
+}
+/**
+ * Get warning that shows layers that will be removed if nofication is accepted
+ * @param {Array} nonDownloadableLayers
+ *
+ * @return {String}
+ */
+export function getNonDownloadableLayerWarning(nonDownloadableLayer) {
+  const layerStr = getNamesOfNondownloadableLayers(nonDownloadableLayer);
+  if (!layerStr) return '';
+  const multiLayers = layerStr.indexOf(',') > -1;
+  return multiLayers ? `The ${layerStr} layers cannot be used to take a snapshot. Would you `
+  + 'like to temporarily hide these layers?' : `The ${layerStr} layer cannot be used to take a snapshot. Would you `
+  + 'like to temporarily hide this layer?';
```

Comment:
```suggestion
  const layerPluralStr = multiLayers ? 'layers' : 'layer';
  return `The ${layerStr} ${layerPluralStr} cannot be included in a snapshot. Would you like to temporarily hide this layer?`;
```

Reply:
Suggested a tweak to the verbiage here to "cannot be included in a snapshot" rather than "cannot be used to take a snapshot"